### DISCUSSION TITLE
Replacement for avoiding layers given that get_route_astar is deprecated

### DISCUSSION THREAD

**Discussion by Mario1159** [gdsfactory/gdsfactory#2478] [Q&A] [answered]
Is there a replacement for routing avoiding certain layers?

**Comment by joamatab** [comment, marked as answer]
Get_route or bundle from steps 


https://gdsfactory.github.io/gdsfactory/_autosummary/gdsfactory.routing.get_bundle_from_steps.html#gdsfactory.routing.get_bundle_from_steps

**Reply by juanferv** [reply]
but is it possible to avoid layers to do the routing? because the way to define the get_route not allow to define avoid_layers and separation, no?

**Reply by joamatab** [reply]
Hi Juan,

we are working on that for the gdsfactory 8.0.0 major release
